Mathlib/Data/Fintype/Pi.lean

```diff
@@ -109,3 +110,49 @@ theorem Finset.univ_pi_univ {α : Type*} {β : α → Type*} [DecidableEq α] [F
     (Finset.univ.pi fun a : α => (Finset.univ : Finset (β a))) = Finset.univ := by
   ext; simp
 #align finset.univ_pi_univ Finset.univ_pi_univ
+
+lemma Fin.succ_mem_piFinset_iff {n : ℕ} {α : Fin (n + 1) → Type*} (p : ∀ i, α i)
+  (S : ∀ i, Finset (α i)) :
```

Comment:
I think `succ` should appear later in the name, and we should use the new notation for dependent functions (which I find much easier to read). Also we should consistent intent by 4 spaces.
```suggestion
lemma Fin.mem_piFinset_succ_iff {n : ℕ} {α : Fin (n + 1) → Type*} (p : (i : Fin (n + 1)) → α i)
    (S : (i : Fin (n + 1)) → Finset (α i)) :
```
(and similarly in various lemmas below)